lms/djangoapps/courseware/views/views.py

```diff
@@ -1079,6 +1082,9 @@ def _get_cert_data(student, course, enrollment_mode, course_grade=None):
     Returns:
         returns dict if course certificate is available else None.
     """
+    # Change the course mode if the student is whitelisted.
+    change_course_mode(course, student)
```

Comment:
Do we have any idea why is this necessary? I has caused problems before, I'm going to test it carefully, we shouldn't merge this yet.

See https://3.basecamp.com/3966315/buckets/6958403/todos/1963631335

Reply:
If this is not added the whitelisted students would not get verified. But I agree, we should review this carefully.